File: benchmarks/concurrency/workers_100.py
import time, threading
results = [0] * 100
def worker(idx):
    s = 0
    for i in range(10000):
        s += i
    results[idx] = s
start = time.monotonic_ns()
threads = [threading.Thread(target=worker, args=(i,)) for i in range(100)]
for t in threads: t.start()
for t in threads: t.join()
end = time.monotonic_ns()
print((end - start) // 1000)

File: benchmarks/concurrency/workers_100.rb
results = Array.new(100, 0)
start = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
threads = 100.times.map { |idx|
  Thread.new {
    s = 0
    10000.times { |i| s += i }
    results[idx] = s
  }
}
threads.each(&:join)
finish = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
puts(finish - start)
